unit-tests/src/main/scala/scala/DoubleComparisonSuite.scala

```diff
@@ -0,0 +1,323 @@
+
+object DoubleComparisonSuite extends tests.Suite {
+  test("fcmp") {
+    // These were code-generated on the JVM using:
+    //
+    // scala> val values = Seq(("0.0d", 0.0d),
+    //                         ("-0.0d", -0.0d),
+    //                         ("Double.MaxValue", Double.MaxValue),
+    //                         ("Double.MinValue", Double.MinValue),
+    //                         ("Double.NaN", Double.NaN),
+    //                         ("42.0d", 42.0d),
+    //                         ("-42.0d", -42.0d))
```

Comment:
You might want to add infinities, here. `Double.PositiveInfinity` and `Double.NegativeInfinity`.